### DISCUSSION TITLE
How to compile and deploy a third-party contract?

### DISCUSSION THREAD

**Discussion by Jeiwan** [NomicFoundation/hardhat#2169] [Troubleshooting] [answered]
Hello!

In a test, I want to deploy the ERC20 contract from `@openzeppelin/contracts`. How can I do that via `ethers.getContractFactory()` without creating a local contract that inherits from the ERC20 contract? In other words, how to compile that contract and load its artifact via `ethers.getContractFactory()`?

Thanks

**Comment by Jeiwan** [comment, marked as answer]
Ended up creating empty dependency-contracts that only import third-party contracts. This make Hardhat compile them so they can be used in tests.

**Reply by fvictorio** [reply]
Hey @Jeiwan, sorry for not answering before.

You are correct that that's a possible solution. Another solution is to use the [hardhat-dependency-compiler plugin](https://www.npmjs.com/package/hardhat-dependency-compiler)

Check [this answer](https://ethereum.stackexchange.com/a/114377/19897) for a more detailed explanation.

**Reply by Jeiwan** [reply]
Thanks!
